tests/trainer/test_trainer.py

```diff
@@ -873,6 +873,52 @@ def on_keyboard_interrupt(self, trainer, pl_module):
     assert isinstance(handle_interrupt_callback.exc_info[1], KeyboardInterrupt)
 
 
+def test_on_exception_hook(tmpdir):
+    """Test the on_exception callback hook."""
```

Comment:
can we combine this and the above test?
It's not the prettiest test I know (I wrote this)

Reply:
Done!